x-pack/plugins/maps/public/classes/layers/solution_layers/security/security_index_pattern_utils.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+/* eslint-disable @typescript-eslint/consistent-type-definitions */
+
+import minimatch from 'minimatch';
+import { SimpleSavedObject } from 'src/core/public';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { IndexPatternSavedObjectAttrs } from 'src/plugins/data/common/index_patterns/index_patterns/index_patterns';
+import { getIndexPatternService, getUiSettings } from '../../../../kibana_services';
+
+export type IndexPatternMeta = {
+  id: string;
+  title: string;
+};
+
+let indexPatterns: IndexPatternMeta[];
+let indexPatternsPromise: Promise<IndexPatternMeta[]>;
+
+export async function getSecurityIndexPatterns(): Promise<IndexPatternMeta[]> {
+  if (indexPatterns) {
+    return indexPatterns;
+  }
+
+  if (!indexPatternsPromise) {
```

Comment:
I'm not sure if we need to write a caching layer on top of the caching already implemented in the IndexPatternService (https://github.com/elastic/kibana/blob/85ebaed37cdf4e1393d967be94b5d6e022215222/src/plugins/data/common/index_patterns/index_patterns/index_patterns.ts#L171-L176).

This seems mainly to avoid having this create that temporary array of `{id, title}` objects so we can just do a `.length > 0` check here: https://github.com/elastic/kibana/blob/853d6787407a90cdf06e4efb2d136a0fce0258b8/x-pack/plugins/maps/public/classes/layers/solution_layers/security/security_layer_wizard.tsx#L16

Remove the duplicate caching and temp-variables associated with it. If we really want to avoid the creation of the temp `{id, title}` objects in `checkVisiblity`, then actually write a visibility check for _that_ specifically.